docs/adr/0033-content-server-upgrade-to-react.md

```diff
@@ -0,0 +1,140 @@
+- **Option D: A single experiment with config settings, and no `/beta` route**: All routes will be guarded by one experiment and separate feature flags per block of routes. In the router file, we will check experiment status as well as feature flag status before send users to the React version of the page via a parameter.
+
+#### Other Considerations
+
+`fxa-content-server` does a lot of things on app startup that Settings does not do, like metrics spin up, recognizing relying parties, OAuth and Sync checks, etc. Because we intend on eventually removing `fxa-content-server`, we need Settings to eventually do these things on app startup instead. When this is implemented, we won't need to access query params provided by content-server.
+
+We may have a follow up ADR to cover the approach to launch `fxa-settings` as the new "base" application depending on complexity and discussions at that time.
+
+### GQL Usage
+
+#### When to Use
+
+We want to extend the GQL functionality that already exists in `fxa-settings` due to the advantages GQL offers (see decision outcomes) and currently, FxA has only implemented GQL on authenticated pages.
+
+- **Option A: Use GQL only on authenticated routes**: Use GQL on authenticated routes, and continue using REST for API calls when users are unauthenticated
+- **Option B: Use GQL everywhere we can**: Add an option per mutation or query to talk to the `graphql-api` without being authenticated, allowing us to use it everywhere **except for likely oauth routes**, which could be another can of worms.
+
+#### Other Implications
+
+If we want to use GQL everywhere we can instead of hitting auth-server directly with REST API calls, and likely even if we're more selective, we have to ensure that the `graphql-api` can handle:
+
+1. Emitting metrics
+2. Sending emails
+3. Sending push notifications (SQS for event-broker)
+
+- **Option I: Create a new API / set of endpoints in `fxa-auth-server` for `graphql-api`**
+  - Pros: faster to implement
+  - Cons: because it’s more ideal to move functionality into a shared space instead, we’re kicking the can down the road and will eventually need to remove the newly created API in favor of using the shared services
+- **Option II: Move these pieces of functionality into stand-alone library-type services in `fxa-shared` with Typescript**: We can use something like [Bull](https://optimalbits.github.io/bull/) or a job/worker scheme to decouple the service that needs to send an email from the actual email sending service
+  - Pros: this gets us closer to our preferred architecture in the long-term and mitigates more refactoring work later
+  - Cons: Large scope increase for this project
+- **Option III: Combination of I + II**: Depending on level of effort and scope increase, only port over 1-2 of these features into `fxa-shared`. Create the new temporary API for other features to be refactored later.
```

Comment:
In my opinion this option will allow us the greatest flexibilty. However, instead of creating a temporary API, we can issue calls using `auth-client`. Once the feature has moved to fxa-shared we can then migrate the graphql route. By doing this, we won't block any frontend work.

Reply:
Part of this also depends on what we consider the role of gql-api to be in FxA.  I think of it as a backend-for-the-frontend service that hand off work to other services.  From that perspective, it is a bit incongruent to rebuild part of the auth-server in gql-api.

Reply:
Hmm yeah, I like the idea of using `auth-client` temporarily as well, that seems preferable over a temporary API. Maybe we can start by building out the front-end in Storybook with l10n and tests, and depending on back-end progress, we can start building out some pages using `auth-client` at that point. I like that it can make some pages more functional faster, but the state management might differ if we're going to use Apollo Cache like we are in Settings and it might not be ideal to backtrack/redo state management and tests. 🤔